.circleci/config.yml

```diff
@@ -478,7 +478,8 @@ jobs:
           name: Install NPM dependencies
           command: |
             npm --version
-            npm install
+            npm ci
+            bundle exec rake npm:check
```

Comment:
Shouldn't we run `npm ls` before `npm install`? And only install in case there are missing packages?

Reply:
`npm ci` removes the node_modules directory, so nothing is "installed". Thus, after ci we check the dependencies were installed correctly... does it makes sense?

Reply:
What I mean is, instead of always running `npm ci`, we should first check if we're missing any dependencies, no? And only install when there are. This is at least what @hallelujah has been suggesting and I thought that's what we were doing.

**Besides that** and according to your comment, why do you think it is necessary to check for missing deps after `npm ci`? The expected behavior is to install everything, maybe it would make more sense to check for output errors from `npm ci`. Am I missing something here?

Reply:
True, it was talked before that better to check first before install. Will change the task now

Reply:
Sadly `npm ci` does not install all dependencies so `npm ls || npm ci` will always run `npm ci` at the end?

Reply:
yeah, that's true... sadly it will ignore the dev deps... so it will fail... 
Will try to do a different approach..

Reply:
The current behavior, theoretically, is that circle ci will only run the NPM task if `package-lock.json` has changed. Otherwise it loads node_modules from the cache.

Isn't that enough?

Reply:
What it was happening, is that sometimes it failed to install all dependencies correctly.. or the cached ones were somehow corrupted. That's why we needed the `npm ls`- you could see the issue attached in the description.
So, no, it's not enough. We need to make sure the dependencies are all there and healthy.

Reply:
OK sorry I didn't read the issue first. I left you a comment in the JIRA, if the issue is missing peer dependencies then it has nothing to do with `npm ci` or `npm ls`.

Reply:
exactly, it's because the local `npm install` and endless package-lock.json modifications. The error generates in dev environment where the only way to spot the missing deps are in the possible warnings of the `npm install` output